### ISSUE TITLE
Hide natlink timer?

### ISSUE THREAD

**Issue by dougransom** [dictation-toolbox/natlinkcore#40]
so grammars don't call unitentnialllyto replace timer callback

**Comment by Danesprite** [comment]
Do you mean the low-level Natlink function (`setTimerCallback`)?

**Comment by dougransom** [comment]
I think so, Quintijn has been working on a multiplexed timer and i think it will get confused if some other code sets the timer callback.  

I think Quintijn's timer can replace the natlink timer.

**Comment by Danesprite** [comment]
Okay then.  There are a few Natlink functions that can be used to break natlinkcore functionality.  `SetTimerCallback` is one, `SetChangeCallback` is another.  This has been the case for some time.  I don't know if it's really worth changing the lower-level API over this.  It's up to you, but I would opt for warnings in the documentation instead.

Dragonfly currently uses the `setTimerCallback` function for its multiplexing timer.  Once *natlinktimer.py* is released, it should be a simple matter to adjust the library to use it instead.  I'll leave the old method in as a fallback for older versions of Natlink.

**Comment by quintijn** [comment]
The natlinktimer.py module directly calls setTimerCallback. When another module calls this same function, the natlinktimer multiplexing will be stopped. I will make a test, to make sure it does.

**Comment by quintijn** [comment]
Dane, I propose to close this. When either Dragonfly of Unimacro uses this, it goes it's own way. 

When Dragonfly and Unimacro try both, we will get conflicts, one of the timers is stopped. Never heard of the case... In principle Dragonfly could adopt the natlinktimer.py of naltinkcore. I think heavier testing should be done before you could decide to go that way.

**Comment by drmfinlay** [comment]
Agreed.  I will adjust Dragonfly to use your natlinktimer module instead at some point.  I would only add that the documentation for `setTimerCallback' should include a warning about overrides.

The documentation for the natlink module could use some work, actually.  I will open a separate issue on that.
